src/library/list.lisp

```diff
@@ -411,3 +411,19 @@
       (concatMap f m))
     (define (>> a b)
       (>>= a (fn (_) b)))))
+
+(cl:defun cl-list-to-coalton (list)
+  (cl:if list
+         (Cons (cl:car list) (cl-list-to-coalton (cl:cdr list)))
+         Nil))
+
+(cl:defun coalton-to-cl-list (list)
```

Comment:
This can be written more simply like (approx pseudocode):

```
(defun coalton-to-cl-list (list)
  (loop :until (typep list 'List/Nil) :collect (prog1 (get-first list) (setf list (rest list))))
```

Reply:
CL-USER> 
(defun coalton-to-cl-list (list)
  (loop :for xs := list :then (slot-value xs 'coalton-library::_1)
        :until (typep xs 'coalton-library::list/nil)
        :collect (slot-value xs 'coalton-library::_0)))
COALTON-TO-CL-LIST

CL-USER> (coalton-to-cl-list (coalton-library:make-list 1 2 3 4))
(1 2 3 4)